FEXCore/Source/Interface/Core/OpcodeDispatcher/Vector.cpp

```diff
@@ -1262,6 +1262,135 @@ OrderedNode* OpDispatchBuilder::SHUFOpImpl(OpcodeArgs, size_t ElementSize,
+          auto DupSrc2 = _VDupElement(DstSize, ElementSize, Src2Node, (Shuffle >> 4) & 0b11);
+          return _VZip(DstSize, 8, Src1Node, DupSrc2);
+        }
+        case 0b00'00'11'10:
+        case 0b01'01'11'10:
+        case 0b10'10'11'10:
+        case 0b11'11'11'10: {
+          // Top elements duplicated, Top 64-bits inserted
+          auto DupSrc2 = _VDupElement(DstSize, ElementSize, Src2Node, (Shuffle >> 4) & 0b11);
+          return _VZip2(DstSize, 8, Src1Node, DupSrc2);
+        }
+        case 0b01'00'01'11: {
+          // TODO: This doesn't generate optimal code.
+          // RA doesn't understand that Src1Node is dead after VInsElement due to SRA class differences.
+          // With RA fixes this would be 2 instructions.
+          // Odd elements inverted, Low 64-bits inserted
+          Src1Node = _VInsElement(DstSize, 4, 0, 3, Src1Node, Src1Node);
+          return _VZip(DstSize, 8, Src1Node, Src2Node);
+        }
+        case 0b11'10'01'11: {
+          // TODO: This doesn't generate optimal code.
+          // RA doesn't understand that Src1Node is dead after VInsElement due to SRA class differences.
+          // With RA fixes this would be 2 instructions.
+          // Odd elements inverted, Top 64-bits inserted
+          Src1Node = _VInsElement(DstSize, 4, 0, 3, Src1Node, Src1Node);
+          return _VInsElement(DstSize, 8, 1, 1, Src1Node, Src2Node);
+        }
+        case 0b01'00'00'01: {
+          // Lower 32-bit elements inverted, low 64-bits inserted
+          Src1Node = _VRev64(DstSize, 4, Src1Node);
+          return _VZip(DstSize, 8, Src1Node, Src2Node);
+        }
+        case 0b11'10'00'01: {
+          // TODO: This doesn't generate optimal code.
+          // RA doesn't understand that Src1Node is dead after VInsElement due to SRA class differences.
+          // With RA fixes this would be 2 instructions.
+          // Lower 32-bit elements inverted, Top 64-bits inserted
+          Src1Node = _VRev64(DstSize, 4, Src1Node);
+          return _VInsElement(DstSize, 8, 1, 1, Src1Node, Src2Node);
+        }
+        case 0b00'00'00'00:
+        case 0b00'00'01'01:
+        case 0b00'00'10'10:
+        case 0b00'00'11'11:
+        case 0b01'01'00'00:
+        case 0b01'01'01'01:
+        case 0b01'01'10'10:
+        case 0b01'01'11'11:
+        case 0b10'10'00'00:
+        case 0b10'10'01'01:
+        case 0b10'10'10'10:
+        case 0b10'10'11'11:
+        case 0b11'11'00'00:
+        case 0b11'11'01'01:
+        case 0b11'11'10'10:
+        case 0b11'11'11'11:
+        {
+          // Duplicate element in upper and lower across each 64-bit segment.
+          auto DupSrc1 = _VDupElement(DstSize, ElementSize, Src1Node, Shuffle & 0b11);
+          auto DupSrc2 = _VDupElement(DstSize, ElementSize, Src2Node, (Shuffle >> 4) & 0b11);
+          return _VZip(DstSize, 8, DupSrc1, DupSrc2);
+        }
+        default:
+          // Intentional fallthrough
```

Comment:
Maybe comment this instead as intentionally doing nothing, since there's no fallthrough from the upper cases.

Reply:
Updated comment.